libraries/AP_Scheduler/AP_Scheduler.cpp

```diff
@@ -389,6 +400,50 @@ void AP_Scheduler::Log_Write_Performance()
     AP::logger().WriteCriticalBlock(&pkt, sizeof(pkt));
 }
 
+// display task statistics as text buffer for @SYS/tasks.txt
+size_t AP_Scheduler::task_info(char *buf, size_t bufsize) const
+{
+    size_t total = 0;
+
+    // a header to allow for machine parsers to determine format
+    int n = snprintf(buf, bufsize, "TasksV1\n");
+    if (n <= 0 || perf_info.get_task_info(0) == nullptr) {
+        return 0;
+    }
+    buf += n;
+    bufsize -= n;
+    total += n;
+
+    uint8_t i = 0;
+
+    do {
+        const AP_Scheduler::Task& task = (i < _num_unshared_tasks) ? _tasks[i] : _common_tasks[i - _num_unshared_tasks];
+        const AP::PerfInfo::TaskInfo* ti = perf_info.get_task_info(i);
+
+        unsigned avg = 0;
```

Comment:
We still try to stay away from non-stdint variable types.

Reply:
Yeah that was because I had to immediately cast for the printf  - but I can change